kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/LdapAuthenticationProviderImpl.scala

```diff
@@ -41,47 +49,72 @@ class LdapAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticat
    * @throws AuthenticationException When a user is found to be invalid by the implementation
    */
   override def authenticate(user: String, password: String): Unit = {
+
+    val (usedBind, bindUser, bindPassword) = (
+      conf.get(KyuubiConf.AUTHENTICATION_LDAP_BIND_USER),
+      conf.get(KyuubiConf.AUTHENTICATION_LDAP_BIND_PASSWORD)) match {
+      case (Some(_bindUser), Some(_bindPw)) => (true, _bindUser, _bindPw)
+      case _ =>
+        // If no bind user or bind password was specified,
+        // we assume the user we are authenticating has the ability to search
+        // the LDAP tree, so we use it as the "binding" account.
+        // This is the way it worked before bind users were allowed in the LDAP authenticator,
+        // so we keep existing systems working.
+        (false, user, password)
+    }
+
+    var search: DirSearch = null
+    try {
+      search = createDirSearch(bindUser, bindPassword)
+      applyFilter(search, user)
+      if (usedBind) {
+        // If we used the bind user, then we need to authenticate again,
+        // this time using the full user name we got during the bind process.
+        createDirSearch(search.findUserDn(user), password)
```

Comment:
@pan3793, I want to support has domain case. Does the authenticate again process needs to get the username part in the provided user?  If yes, I will create the function `getUserName` and provide a PR.
`  def getUserName(user: String): String =
    if (LdapUtils.hasDomain(user)) LdapUtils.extractUserName(user) else user`

Reply:
` val username = getUserName(user)
   createDirSearch(search.findUserDn(username), password)`

Reply:
Sounds make sense to me. The current logic should be same as Apache Hive, would you please check if Hive has same issue? Anyway, go ahead to improve it.

Reply:
sure, Hive branch 4.0.0 has the same issue, I will improve it.